core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/operationresult/CmSyncOperationResultEvaluatorService.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.cloudbreak.service.upgrade.sync.operationresult;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.service.upgrade.sync.common.ParcelInfo;
+
+@Service
+public class CmSyncOperationResultEvaluatorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CmSyncOperationResultEvaluatorService.class);
+
+    public CmSyncOperationSummary.Builder evaluateParcelSync(CmParcelSyncOperationResult cmParcelSyncOperationResult) {
+        if (cmParcelSyncOperationResult.getInstalledParcels().isEmpty()) {
+            String message = "CM parcel sync failed, it was not possible to retrieve installed parcel versions from the CM server.";
+            LOGGER.debug(message);
+            return CmSyncOperationSummary.builder().withError(message);
+        }
```

Comment:
could you add an else here and move the whole else's body to a method?

Reply:
done